py3status/modules/sysdata.py

```diff
@@ -267,13 +264,36 @@ def _get_cputemp(self, zone, unit):
         m = re.search("(Core 0|CPU Temp).+\+(.+).+\(.+", sensors)
         if m:
             cpu_temp = float(m.groups()[1].strip()[:-2])
-            if unit == "K":
-                cpu_temp += 273.15
         else:
             cpu_temp = "?"
 
         return cpu_temp
 
+    def _get_cputemp(self, zone, unit):
+        # if unit not in [u"°F", u"°C", "K"]:
+        #     return "unknown unit"
+        if zone is not None:
+            try:
+                with open(zone) as f:
+                    cpu_temp = f.readline()
+                    cpu_temp = float(cpu_temp) / 1000  # convert from mdegC to degC
+            except (OSError, IOError, ValueError):
+                # FileNotFoundError does not exist on Python < 3.3, so we catch OSError instead
+                # ValueError can be thrown if zone was a file that didn't have a float
+                # if zone was not a file, it might be a sensor!
+                cpu_temp = self._get_cputemp_with_lmsensors(zone=zone)
+
+        else:
+            cpu_temp = self._get_cputemp_with_lmsensors()
+
+        if cpu_temp is float:
+            if unit == u"°F":
+                cpu_temp = cpu_temp * (9 / 5) + 32
```

Comment:
be careful with this, on python2 `9 / 5` returns `1`... Either just write `1.8`, or use explicit floats `9.0 / 5.0`

Reply:
Changed to explicit floats